### PR TITLE
GH-2081 reorg sparql custom compliance tests as JUnit 4 Test suite

### PR DESCRIPTION
**Pull request by jeenbroekstra**
GitHub issue resolved: #2081  

Briefly describe the changes proposed in this PR:

- reorganized ComplexSPARQLQueryTest class into a suite of several test classes
- using JUnit 4 Suite setup as I could not get things working with the JUnit 5 `@Suite` annotation
- rough separation into related test cases, can be refined/extended further as necessary

----
PR Author Checklist (see the [contributor guidelines](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md) for more details):

 - [x] my pull request is [self-contained](https://rdf4j.org/documentation/developer/merge-strategy/#self-contained-changes-pull-requests-and-commits)
 - [x] I've added tests for the changes I made
 - [x] I've applied [code formatting](https://github.com/eclipse/rdf4j/blob/main/.github/CONTRIBUTING.md#code-formatting) (you can use `mvn process-resources` to format from the command line)
 - [x] I've [squashed](https://rdf4j.org/documentation/developer/squashing) my commits where necessary 
 - [x] every commit message starts with the issue number (GH-xxxx) followed by a meaningful description of the change

### CODE REVIEW

**Comment by hmottestad** [comment]
I was wondering about doing some rewrites of those tests but was wondering if they are not published on maven? In which case others might be using them.

**Comment by jeenbroekstra** [comment]
> I was thinking about doing some rewrites of those tests but was wondering if they are not published on maven? In which case others might be using them.

That's a good point - I may leave the old set of tests in (deprecated) purely for backward compatibility. They won't run if we don't have concrete implementations on our end, but other parties can continue using them.

**Comment by JervenBolleman** [comment]
Maybe ask around if anyone does? people can always continue to use the old test jars.

**Comment by jeenbroekstra** [comment]
> Maybe ask around if anyone does? people can always continue to use the old test jars.

Unfortunately I've found in the past that "asking around" is not a very reliable system. People don't pay attention to that kind of stuff until they bump versions and suddenly stuff breaks.

In any case leaving the old tests in places in deprecated form is not a big chore, it's a single Java class after all :)

**Comment by JervenBolleman** [comment]
> > Maybe ask around if anyone does? people can always continue to use the old test jars.
> 
> Unfortunately I've found in the past that "asking around" is not a very reliable system. People don't pay attention to that kind of stuff until they bump versions and suddenly stuff breaks.
> 
> In any case leaving the old tests in places in deprecated form is not a big chore, it's a single Java class after all :)

Ok, maybe add a `@Deprecated(forRemoval=true)` so we can delete it at version 5 ?

**Comment by jeenbroekstra** [comment]
Updated this, restoring the old suite as a deprecated class, and improved the javadoc a bit here and there.

File: testsuites/sparql/pom.xml [hunk 1]
```diff
@@ -96,6 +96,11 @@
 			<artifactId>rdf4j-repository-contextaware</artifactId>
 			<version>${project.version}</version>
 		</dependency>
+		<dependency>
+			<groupId>${project.groupId}</groupId>
+			<artifactId>rdf4j-repository-manager</artifactId>
+			<version>${project.version}</version>
+		</dependency>
```
Comment (jeenbroekstra) [inline testsuites/sparql/pom.xml:103]: Ugh - I should take this out again, I put this in to try something but went with a different approach instead.

**Comment by hmottestad** [comment]
Could it be useful to mark the new classes as experimental or internal only so that we can keep improving them and maybe move them to junit 5?

**Comment by jeenbroekstra** [comment]
> Could it be useful to mark the new classes as experimental or internal only so that we can keep improving them and maybe move them to junit 5?

Good point, will do.

**Pull request merged**